### ISSUE TITLE
with `mask` output is `nan`

### ISSUE THREAD

**Issue by DrHB** [lucidrains/x-transformers#27]
I am trying to run image captioning example with the `mask` for caption. Without `mask` forward pass runs properly but with `mask` I get the `nan`. You can reproduce by running following simple example. 

```
import torch
from x_transformers import ViTransformerWrapper, TransformerWrapper, Encoder, Decoder

encoder = ViTransformerWrapper(
    image_size = 256,
    patch_size = 32,
    attn_layers = Encoder(
        dim = 512,
        depth = 6,
        heads = 8
    )
)

decoder = TransformerWrapper(
    num_tokens = 20000,
    max_seq_len = 1024,
    attn_layers = Decoder(
        dim = 512,
        depth = 6,
        heads = 8,
        cross_attend = True
    )
)

img = torch.randn(1, 3, 256, 256)
caption = torch.randint(0, 20000, (1, 1024))
mask = torch.ones_like(caption, dtype=torch.bool)
encoded = encoder(img)
decoder(caption, context = encoded, mask = mask) 
```

output:

```
tensor([[[nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         ...,
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan],
         [nan, nan, nan,  ..., nan, nan, nan]]], grad_fn=<UnsafeViewBackward>)
```

**Comment by lucidrains** [comment]
@DrHB hmm that's weird, i'm running your example and i see

```bash
tensor([[[-0.4690, -0.1911, -0.3346,  ..., -0.5400, -0.2755,  0.9568],
         [-0.2044, -0.1291, -0.0293,  ..., -0.5013, -0.7330,  0.7467],
         [-0.2840,  0.3205, -0.2971,  ..., -0.7514, -0.6133,  0.6637],
         ...,
         [ 0.3447, -0.6681, -0.8066,  ..., -1.1617, -0.4271, -0.5544],
         [ 0.2771, -0.7323, -0.8410,  ..., -1.0202, -0.3535, -0.5067],
         [ 0.3569, -0.7768, -0.7122,  ..., -0.9473, -0.4585, -0.6757]]],
       grad_fn=<AddBackward0>)
```

**Comment by lucidrains** [comment]
```bash
tensor([[[-1.0632,  0.2720,  0.1282,  ...,  0.2447, -0.7386,  0.3464],
         [-0.8407,  0.1912,  0.1427,  ...,  0.3989, -0.9206,  0.4414],
         [-0.5831, -0.0275,  0.2718,  ...,  0.2846, -0.9695,  0.4377],
         ...,
         [ 1.2271,  1.1882,  0.2487,  ...,  0.4520, -0.8317,  0.6442],
         [ 1.1818,  1.1907,  0.1437,  ...,  0.4029, -0.7247,  0.7719],
         [ 1.1336,  1.1832,  0.1195,  ...,  0.4593, -0.6084,  0.6281]]],
       grad_fn=<AddBackward0>)
```

hmm i get an output - what version of x-transformers are you on? (and version of pytorch, etc)

**Comment by DrHB** [comment]
Pytorch version `1.7.1+cu101`, 
x_transformer   `x_transformers-0.10.1`

**Comment by DrHB** [comment]
I updated Pytorch. It works now. Thank you.
